Implementation for the Coding Test

pmielke89 opened the pull request:
I just finished my coding test, so here's my implementation. Hope to hear from you again, soon.

Commit:
Setting up the project

Folder structure created, added index.html and scss/css.

Commit:
Header implemented

HTML and SCSS (CSS) of the Header implemented.

Commit:
Vue instance implemented and first test data created

Commit:
Started implementation of table

Region level can be displayed, but the view is not yet finished

Commit:
expand-data now has it's own td

All elements of that class are now exactly below each other.

Commit:
Minor improvement in style.scss and index.html

On the region level, expand-data will only be displayed, if there are
districts in the data set. expand-data:after has been added to attach an
"+" at the end of the div.

Commit:
Added region icon to table

Changes for that were made in index.html and style.scss

Commit:
Data can now be displayed up to the township level

Commit:
Started working on expanding/reducing table

District level can now be toggled. Township level is not yet
implemented.

Commit:
Table: Region, District and Township levels can now be toggled

Commit:
Started implementation of search filter

Commit:
Implementation of search filter complete

Commit:
Changed README.md; the state of the toggle button (+/-) is now being changed correctly

Commit:
Test cases implemented

Commit:
Cleaning up and copying files to /dist

Commit:
Updating Readme.md and cleaning files

Commit:
Fixing typos in README.md

Commit:
Fixed another typo

Commit:
Changed some (S)CSS of the table

The tr width is now 45px.

Jamesford commented:
Hi @pmielke89, sorry to be getting back to you so late, hibernating 🐻  over the holidays. Here is some initial feedback and questions for your task.

Firstly its awesome that you have included some tests, but we prefer to not globally install modules. Could you add karma & jasmine as (dev)dependencies via `npm` and set it up so after installing dependencies we could just execute `npm test` to run the tests?

Secondly, what (if any) build tools are you using? At least you are compiling SCSS to CSS, what is your workflow for that task? Have you used [vue-cli](https://github.com/vuejs/vue-cli) before?

Commit:
Added package.json; modified test files

package.json: Karma and Jasmine can now be installed by doing 'npm
install' and the tests can be performed by using 'npm test'.

karma.conf.js: Changed the reporter to 'spec'.

test.js: Modified some specs' names, to clearify, that that methods are
being tested.

pmielke89 commented:
Hey @Jamesford, that's totally okay! Hope you had great holidays! :santa:

Thanks for your first feedback!

I changed the project setup as requested, so you should be able to install the dependencies with `npm install` and then run the tests by executing `npm test`.

The dependencies are:

- `karma`

- `jasmine-core`

- `karma-jasmine`

- `karma-chrome-launcher`

- `karma-spec-reporter`

I've been using Brackets as IDE and there's an extension called Brackets SASS, which can compile SASS/SCSS to CSS. This seemed to be the easiest solution, since I only had this single file to compile. :smiley:

No, unfortunately, I've never worked with `vue-cli` before.
